frontend/src/components/hub/HubSupporter.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useContext } from "react";
+import makeStyles from "@mui/styles/makeStyles";
+import Carousel from "react-multi-carousel";
+import { Theme, useMediaQuery, Link, Typography } from "@mui/material";
+import UserContext from "../context/UserContext";
+import { getImageUrl } from "../../../public/lib/imageOperations";
+import getTexts from "../../../public/texts/texts";
+import ArrowRightIcon from "@mui/icons-material/ArrowRight";
+
+type Supporter = {
+  name: string;
+  subtitle: string;
+  logo: string;
+  importance: number;
+};
+
+type HubSupporter = {
+  supportersList: Supporter[];
+  containerClass?: string;
+};
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    backgroundColor: "#207178",
+    borderRadius: 4,
+    paddingBottom: "15px",
+    paddingRight: "5px",
+    paddingLeft: "5px",
+    width: 320,
+    //instead of width we can use flexGrow && maxWidth
+    // flexGrow: 1,
+    // maxWidth: 320
+  },
+  carouseltitle: {
+    color: "#FFFFFF",
+    fontSize: "13px",
+    margin: "2px",
+    textAlign: "center",
+  },
+  carouselContainer: {
+    backgroundColor: "#F0F2F5",
+    borderRadius: "4px",
+    position: "relative",
+  },
+  customDot: {
+    bottom: "-16px",
+    // access the class name of the react-multi-carousel to change the dot color
+    "& .react-multi-carousel-dot--active button": {
+      background: "white",
+    },
+    "& li button": {
+      width: "7px",
+      height: "7px",
+      border: "none",
+      background: "#66BCB5",
+    },
+  },
+
+  itemContainer: {
+    display: "flex",
+    alignItems: "center",
+  },
+  supporterImg: {
+    borderRadius: "50%",
+  },
+  supporterDetails: {
+    paddingLeft: "15px",
+  },
+  supporterName: {
+    fontSize: "17px",
+    fontWeight: "600",
+    marginBottom: "8px",
+  },
+  supporterSubtitle: (containerClass) => ({
+    margin: "0",
+    fontSize: "12px",
+    color: "#484848",
+    // fontWeight: 600,
+    whiteSpace: "nowrap",
+    width: containerClass ? "170px" : "200px",
+    overflow: "hidden",
+    OTextOverflow: "ellipsis",
+    textOverflow: "ellipsis",
+  }),
+  carouselEntry: {
+    padding: " 8px 9px 15px 10px",
+    display: "flex",
+    justifyContent: "left",
+    [theme.breakpoints.down("md")]: {
+      padding: 0,
+    },
+  },
+  containerInSmallDevices: {
+    display: "flex",
+    gap: "20px",
+    alignItems: "center",
+    backgroundColor: "#EEEFEEE8",
+    borderRadius: "4px",
+    padding: "10px",
+    marginBottom: theme.spacing(3),
+    marginTop: theme.spacing(2),
+  },
+  supporterImgSmallDevice: {
+    borderRadius: "50%",
+  },
+  textAlign: {
+    marginLeft: "auto",
+  },
+  allSupporters: {
+    display: "flex",
+    alignItems: "center",
+    color: "#484848",
+    fontWeight: "600",
+    fontSize: "17px",
+  },
+  arrowIcon: {
+    color: "#207178",
+  },
+}));
+
+const HubSupporter = ({ supportersList, containerClass }: HubSupporter) => {
```

Comment:
I've changed the Front component name and variables.
What about the model name in the backend? 
Currently, it’s named `HubSupporter`. Do you think it would be better to use a plural form, like `HubSupporters` ?

Reply:
@fateme-ramezanpour No, in the backend the naming is fine :) It's just about the frontend components